File: examples/javascript/queue_1.js
class Queue {
  constructor() { this.items = []; }
  enqueue(val) { this.items.push(val); }
  dequeue() { return this.items.shift(); }
  front() { return this.items[0]; }
  isEmpty() { return this.items.length === 0; }
}

const queue = new Queue();
queue.enqueue(5); queue.enqueue(10); queue.enqueue(15);
console.log(queue.dequeue());
console.log(queue.front());
console.log(queue.dequeue());

File: examples/ruby/queue_1.rb
class Queue
  def initialize
    @data = []
  end

  def enqueue(x)
    @data << x
  end

  def dequeue
    @data.empty? ? nil : @data.shift
  end
end

q = Queue.new
q.enqueue(1)
q.enqueue(2)
q.enqueue(3)
puts q.dequeue
puts q.dequeue
